nav2_dwb_controller/nav_2d_utils/include/nav_2d_utils/odom_subscriber.hpp

```diff
@@ -78,13 +89,31 @@ class OdomSubscriber
     std::lock_guard<std::mutex> lock(odom_mutex_);
     odom_vel_.header = msg->header;
     odom_vel_.velocity.x = msg->twist.twist.linear.x;
-    odom_vel_.velocity.y = msg->twist.twist.linear.y;
+    if (robot_type_ == HOLONOMIC) {
+      odom_vel_.velocity.y = msg->twist.twist.linear.y;
+    } else {
+      odom_vel_.velocity.y = 0;
```

Comment:
I really think this has to do with an issue in the turtlebot code that 100% needs to be resolved. You should never be receiving odometry from a robot that can only move forward and rotate in the Y axis in reality. If Gazebo is giving you back this information, something is terribly wrong. 

This is masking a much larger issue and I think its more dangerous than helpful for the general user. I have to assume that if there's a Y component even though there shouldn't be, there's some component of the odometry in Y that is actually to be projected into X and these values are therefore screwy somehow. 

Patches for simulations probably don't belong on production hardware code. Maybe its a better choice to fork the gazebo diff drive plugin and force it 0 there or have the turtlebot3 org fix the underlying issues that results in this

Reply:
*oh oh oh oh* I mentioned in another ticket that I'd run into this before, so I racked my brain and remember I posted on ROS Answers, here's the thread: https://answers.ros.org/question/283919/move-base-publishing-y-velocities/

Its TF! Something's not publishing enough so there's interpolations happening that incorrectly assign the transforms (and/or footprint)

Reply:
I looked into that some. There could be a TF problem. The dimensions of the robot used by Gazebo don't match those used by `robot_state_publisher`. Gazebo uses the SDF model while `robot_state_publisher` uses the URDF model.

But I don't think it is the cause of this problem. I can reproduce with no transforms being published or used in the system. Just Gazebo and teleop are running.